staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -91,8 +50,8 @@ var _ http.Hijacker = &fancyResponseWriterDelegator{}
 // 2. the response line containing:
 //    - the unique id from 1
 //    - response code
-func WithAudit(handler http.Handler, requestContextMapper request.RequestContextMapper, out io.Writer) http.Handler {
-	if out == nil {
+func WithAudit(handler http.Handler, requestContextMapper request.RequestContextMapper, sink audit.Sink, policy *auditinternal.Policy, longRunningCheck filters.LongRunningRequestCheck) http.Handler {
```

Comment:
Per our discussion on https://github.com/kubernetes/kubernetes/pull/46009#discussion_r117443109, can we keep a copy of the old logic around?